packages/react-reconciler/src/ReactFiberLane.js

```diff
@@ -604,6 +622,19 @@ export function findTransitionLane(
   );
 }
 
+// To ensure consistency across multiple updates in the same event, this should
+// be pure function, so that it always returns the same lane for given inputs.
+export function findRetryLane(wipLanes: Lanes): Lane {
+  // This is a fork of `findUpdateLane` designed specifically for Suspense
+  // "retries" — a special update that attempts to flip a Suspense boundary
+  // from its placeholder state to its primary/resolved state.
+  let lane = findLane(RetryRangeStart, RetryRangeEnd, wipLanes);
+  if (lane === NoLane) {
+    lane = pickArbitraryLane(RetryLanes);
```

Comment:
Late to the party here (sorry, oncall this week and a bit behind) but catching up now, I had a small thought.

The naming of `pickArbitraryLane` suggests (to me) that it could return a different value each time it's called. Its current implementation always happens to return a predictable/consistent value, but do we think this might ever change? If it did, the usage of it would be at conflict with the above comment:
> This should be pure function, so that it always returns the same lane for given inputs.

...in which case maybe calling `getLowestPriorityLane` directly would be "better"

Reply:
“arbitrary” in this case means it doesn’t matter if it’s the lowest priority lane or highest rioting lane or anything in between.

For example, it communicates if there’s a refactor where the most significant lane is available for some other reason, you could use that instead without affecting the algorithm.